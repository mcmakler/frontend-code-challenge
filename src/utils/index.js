import priceFormatter from './price-formatter/price-formatter';
import trim from './trim/trim';
import round from './round/round';
import carder from './carder/carder';
import flow from './flow/flow';
import * as specialCharacter from './special-characters/special-characters';
import * as mapper from './mapper/mapper';

export {
  priceFormatter,
  specialCharacter,
  mapper,
  trim,
  round,
  flow,
  carder
};
