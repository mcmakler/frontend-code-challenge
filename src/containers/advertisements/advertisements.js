import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { loadAdvertisments } from '../../actions/advertisements-actions/advertisements-actions';
import { carder } from '../../utils';

import Page from '../page/page';
import Card from '../../components/card/card';

export class AdvertisementsContainer extends React.Component {
  constructor(props, context) {
    super(props, context);

    const loaded = false;
    this.state = { loaded };
  }

  componentDidMount() {
    this._fetch();
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ advertisements: nextProps.advertisements, loaded: true });
  }

  _fetch() {
    this.props.loadAdvertisments();
  }

  render() {
    return (
      <Page loaded={this.state.loaded}>
        <div className='row center-xs between-md'>
          {
            this.props.advertisements.map(item => {
              const options = carder(item);

              return (<Card key={item.id} {...options} />);
            })
          }
        </div>
      </Page>
    );
  }
}

AdvertisementsContainer.propTypes = {
  loadAdvertisments: PropTypes.func.isRequired,
  advertisements: PropTypes.array.isRequired
};

function mapStateToProps(state) {
  return state;
}

function mapDispatchToProps(dispatch) {
  return {
    loadAdvertisments: () => {
      dispatch(loadAdvertisments());
    }
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(AdvertisementsContainer);
