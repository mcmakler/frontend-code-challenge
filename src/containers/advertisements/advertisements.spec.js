import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {AdvertisementsContainer} from './advertisements';
import {mapper} from '../../utils';
import advertisementObj from '../../../fixtures/advertisement-fixture';

describe('test advertisements', () => {
  it('should render advertisements container', () => {
    const formatted = mapper.itemMapper(advertisementObj);
    const props = {
      loadAdvertisments: () => {},
      advertisements: [formatted]
    };
    const component = shallow(<AdvertisementsContainer {...props}/>);

    expect(toJson(component)).toMatchSnapshot();
  });
});
