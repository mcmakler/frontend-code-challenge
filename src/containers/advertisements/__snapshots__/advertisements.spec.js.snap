// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test advertisements should render advertisements container 1`] = `
<PageContainer
  loaded={false}
>
  <div
    className="row center-xs between-md"
  >
    <Card
      footer={
        Object {
          "items": Array [
            Object {
              "name": "price",
              "options": Object {
                "bold": true,
              },
              "postLabel": "â‚¬",
              "value": "374",
            },
            Object {
              "name": "rooms",
              "options": Object {
                "right": true,
                "seperated": true,
              },
              "postLabel": "Zimmer",
              "value": 3,
            },
            Object {
              "name": "space",
              "options": Object {
                "right": true,
                "seperated": true,
              },
              "postLabel": <span>
                m
                <span
                  className="align-super font-10"
                >
                  2
                </span>
              </span>,
              "preLabel": "ab",
              "value": 79,
            },
          ],
        }
      }
      header={
        Object {
          "alt": "FLATSCREEN ZUM EINZUG Moderne, frisch sanierte 3",
          "tag": "mieten",
          "thumb": "https://d3e02gns9oqhhr.cloudfront.net/assets/expose_v2/573c7c608ffd7c0b2a8b45b6_variant_asset57ffa0ced86b1_inventoryM.jpg",
        }
      }
      key="573c7c608ffd7c0b2a8b45b6"
      text={
        Object {
          "text": "Im Bonnefeld 7, 47259 Duisburg",
        }
      }
      title={
        Object {
          "title": "FLATSCREEN ZUM EINZUG Moderne, frisch sanierte 3",
        }
      }
    />
  </div>
</PageContainer>
`;
